## **Consigna:** 
### Identificar características generales para los trabajos prácticos.
### Resumir esta información en 1-2 diapositivas.






In [ ]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

df = pd.read_csv("../Datasets/ged251-csv/GEDEvent_v25_1.csv")

C:\Users\tolot\AppData\Local\Temp\ipykernel_13784\2609162951.py:6: DtypeWarning: Columns (0: gwnoa) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("../Datasets/ged251-csv/GEDEvent_v25_1.csv")


## Exploración inicial del dataset

Se realiza una inspección rápida del dataset para entender su estructura.  
Se muestran las primeras filas, las dimensiones, los nombres de las columnas y la información general de los datos

In [15]:
from IPython.display import display

print("=== Primeras filas del dataset ===")
display(df.head())

print("\n=== Dimensiones del dataset (filas, columnas) ===")
display(df.shape)

print("\n=== Columnas del dataset ===")
display(df.columns)

print("\n=== Información general ===")
df.info()

=== Primeras filas del dataset ===


,id,relid,year,active_year,code_status,type_of_violence,conflict_dset_id,conflict_new_id,conflict_name,dyad_dset_id,...,date_end,deaths_a,deaths_b,deaths_civilians,deaths_unknown,best,high,low,gwnoa,gwnob
0,244657,IRQ-2017-1-524-322,2017,True,Clear,1,259,259,Iraq: Government,524,...,2017-07-31 00:00:00.000,0,4,0,2,6,6,6,645,NaN
1,412700,IRQ-2021-1-524-145,2021,True,Clear,1,259,259,Iraq: Government,524,...,2021-08-26 00:00:00.000,13,1,141,28,183,184,171,645,NaN
2,413023,IRQ-2021-1-524-143,2021,True,Clear,1,259,259,Iraq: Government,524,...,2021-08-28 00:00:00.000,0,2,0,0,2,3,0,645,NaN
3,412909,IRQ-2021-1-524-144,2021,True,Clear,1,259,259,Iraq: Government,524,...,2021-08-29 00:00:00.000,0,0,10,0,10,10,9,645,NaN
4,132140,AFG-1989-1-411-2,1989,True,Clear,1,333,333,Afghanistan: Government,724,...,1989-01-13 00:00:00.000,6,0,0,0,6,6,6,700,NaN



=== Dimensiones del dataset (filas, columnas) ===


(385918, 49)


=== Columnas del dataset ===


Index(['id', 'relid', 'year', 'active_year', 'code_status', 'type_of_violence',
       'conflict_dset_id', 'conflict_new_id', 'conflict_name', 'dyad_dset_id',
       'dyad_new_id', 'dyad_name', 'side_a_dset_id', 'side_a_new_id', 'side_a',
       'side_b_dset_id', 'side_b_new_id', 'side_b', 'number_of_sources',
       'source_article', 'source_office', 'source_date', 'source_headline',
       'source_original', 'where_prec', 'where_coordinates',
       'where_description', 'adm_1', 'adm_2', 'latitude', 'longitude',
       'geom_wkt', 'priogrid_gid', 'country', 'country_id', 'region',
       'event_clarity', 'date_prec', 'date_start', 'date_end', 'deaths_a',
       'deaths_b', 'deaths_civilians', 'deaths_unknown', 'best', 'high', 'low',
       'gwnoa', 'gwnob'],
      dtype='str')


=== Información general ===
<class 'pandas.DataFrame'>
RangeIndex: 385918 entries, 0 to 385917
Data columns (total 49 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   id                 385918 non-null  int64  
 1   relid              385918 non-null  str    
 2   year               385918 non-null  int64  
 3   active_year        385918 non-null  bool   
 4   code_status        385918 non-null  str    
 5   type_of_violence   385918 non-null  int64  
 6   conflict_dset_id   385918 non-null  int64  
 7   conflict_new_id    385918 non-null  int64  
 8   conflict_name      385918 non-null  str    
 9   dyad_dset_id       385918 non-null  int64  
 10  dyad_new_id        385918 non-null  int64  
 11  dyad_name          385918 non-null  str    
 12  side_a_dset_id     385918 non-null  int64  
 13  side_a_new_id      385918 non-null  int64  
 14  side_a             385918 non-null  str    
 15  side_b_dset_id     385918 non-nul

## Tipos de variables

En este bloque se identifican los tipos de datos presentes en el dataset.  
Esto permite distinguir entre variables numéricas, categóricas, booleanas o de fecha, lo cual es importante para decidir qué análisis y visualizaciones aplicar.

In [9]:
# Mostrar todas las filas
pd.set_option("display.max_rows", None)

print("=== Tipos de datos de cada variable ===")
display(df.dtypes.to_frame(name="Tipo"))

print("\n=== Conteo de tipos de variables ===")
display(df.dtypes.value_counts())

print("\n=== Variables numéricas ===")
display(df.select_dtypes(include=["number"]).columns)

print("\n=== Variables categóricas ===")
display(df.select_dtypes(include=["object", "category", "string"]).columns)

print("\n=== Variables booleanas ===")
display(df.select_dtypes(include=["bool"]).columns)

print("\n=== Variables de fecha ===")
display(df.select_dtypes(include=["datetime"]).columns)

=== Tipos de datos de cada variable ===


,Tipo
id,int64
relid,str
year,int64
active_year,bool
code_status,str
type_of_violence,int64
conflict_dset_id,int64
conflict_new_id,int64
conflict_name,str
dyad_dset_id,int64



=== Conteo de tipos de variables ===


int64      24
str        20
float64     3
bool        1
object      1
Name: count, dtype: int64


=== Variables numéricas ===


Index(['id', 'year', 'type_of_violence', 'conflict_dset_id', 'conflict_new_id',
       'dyad_dset_id', 'dyad_new_id', 'side_a_dset_id', 'side_a_new_id',
       'side_b_dset_id', 'side_b_new_id', 'number_of_sources', 'where_prec',
       'latitude', 'longitude', 'priogrid_gid', 'country_id', 'event_clarity',
       'date_prec', 'deaths_a', 'deaths_b', 'deaths_civilians',
       'deaths_unknown', 'best', 'high', 'low', 'gwnob'],
      dtype='str')


=== Variables categóricas ===


Index(['relid', 'code_status', 'conflict_name', 'dyad_name', 'side_a',
       'side_b', 'source_article', 'source_office', 'source_date',
       'source_headline', 'source_original', 'where_coordinates',
       'where_description', 'adm_1', 'adm_2', 'geom_wkt', 'country', 'region',
       'date_start', 'date_end', 'gwnoa'],
      dtype='str')


=== Variables booleanas ===


Index(['active_year'], dtype='str')


=== Variables de fecha ===


Index([], dtype='str')

## Calidad de datos: duplicados, valores nulos y outliers

Se analizan posibles problemas en la calidad del dataset.
Se identifican registros duplicados, variables con valores faltantes y posibles outliers en variables numéricas mediante el método del rango intercuartílico (IQR).
Solo se muestran las variables que presentan nulos o outliers, lo que permite enfocarse en inconsistencias que podrían afectar el análisis posterior.

In [11]:
# Duplicados
print("=== Registros duplicados ===")
duplicados = df.duplicated().sum()
print("Cantidad de filas duplicadas:", duplicados)

# Valores nulos
print("\n=== Valores nulos por variable (solo > 0) ===")
nulos = df.isnull().sum()
nulos_filtrados = nulos[nulos > 0]  # solo columnas con nulos
display(nulos_filtrados)

# Porcentaje de nulos
print("\n=== Porcentaje de valores nulos (solo > 0%) ===")
porcentaje_nulos = (df.isnull().sum() / len(df)) * 100
porcentaje_nulos_filtrado = porcentaje_nulos[porcentaje_nulos > 0]  # solo > 0%
display(porcentaje_nulos_filtrado)

# Outliers con IQR
print("\n=== Outliers en variables numéricas (solo columnas con outliers) ===")
df_num = df.select_dtypes(include="number")
Q1 = df_num.quantile(0.25)
Q3 = df_num.quantile(0.75)
IQR = Q3 - Q1

outliers = ((df_num < (Q1 - 1.5 * IQR)) | (df_num > (Q3 + 1.5 * IQR)))
outliers_count = outliers.sum()

# Filtrar solo columnas con outliers
outliers_filtrados = outliers_count[outliers_count > 0]
display(outliers_filtrados)

=== Registros duplicados ===
Cantidad de filas duplicadas: 0

=== Valores nulos por variable (solo > 0) ===


source_article            1
source_office         97913
source_date           97913
source_headline       97918
source_original       44244
where_description     10243
adm_1                 19547
adm_2                 69977
gwnoa                 93411
gwnob                355662
dtype: int64


=== Porcentaje de valores nulos (solo > 0%) ===


source_article        0.000259
source_office        25.371452
source_date          25.371452
source_headline      25.372748
source_original      11.464612
where_description     2.654191
adm_1                 5.065066
adm_2                18.132608
gwnoa                24.204883
gwnob                92.159993
dtype: float64


=== Outliers en variables numéricas (solo columnas con outliers) ===


conflict_dset_id      52145
conflict_new_id       90243
side_a_dset_id       102119
side_a_new_id        102119
number_of_sources      5192
where_prec              251
latitude               3473
longitude             54154
priogrid_gid           3461
event_clarity         29603
date_prec             57245
deaths_a              94908
deaths_b              57128
deaths_civilians      40484
deaths_unknown        56401
best                  50210
high                  45916
low                   46614
gwnob                  1757
dtype: int64

# Resumen del Dataset – EDA Inicial

---

## Información General

**Tamaño del dataset:** 385,918 filas × 49 columnas

**Tipos de variables:**

| Tipo       | Cantidad |
|-----------|----------|
| Numéricas | 24       |
| Categóricas / Texto | 20 |
| Flotantes | 3        |
| Booleanas | 1        |
| Objeto    | 1        |

**Variables destacadas:**

- **Numéricas:** `id, year, number_of_sources, latitude, longitude, deaths_a, deaths_b, deaths_civilians, ...`
- **Categóricas:** `conflict_name, dyad_name, side_a, side_b, source_article, source_office, source_date, region, ...`

**Valores nulos más relevantes:**

| Variable           | % Nulos |
|------------------|--------|
| gwnob             | 92%    |
| source_office     | 25%    |
| source_date       | 25%    |
| source_headline   | 25%    |
| adm_2             | 18%    |

**Curiosidad:**  
- Algunas variables numéricas (`deaths_a`, `deaths_b`, `number_of_sources`) presentan una cantidad significativa de outliers.

---

## Desafíos a priori

- **Limpieza de datos:**
  - Columnas con alto porcentaje de nulos (`gwnob`, `source_office`, `source_date`)
  - Outliers en `deaths_a`, `deaths_b`, `side_a_new_id`, `number_of_sources`
  - Revisar posibles duplicados
- **Preparación de variables:**
  - Variables categóricas largas (`source_headline`, `source_article`) requieren limpieza
  - Evaluar redundancia entre IDs (`conflict_dset_id` vs `conflict_new_id`)
- **Otros retos:**
  - Escalabilidad del análisis (~386k filas)
  - Consistencia entre IDs numéricos y nombres categóricos (`side_a`, `side_b`, `dyad_name`)

---

💡 *Tips visuales para la presentación:*  
- Gráfico de barras o pastel: porcentaje de nulos por variable más relevante  
- Boxplots: variables numéricas con outliers

# Análisis del dataset seleccionado

---

## Características generales del dataset

- **Tamaño:** 385,918 filas × 49 columnas
- **Tipos de variables:**
  - Numéricas: 24
  - Categóricas / texto: 20
  - Flotantes: 3
  - Booleanas: 1
  - Objeto: 1
- **Variables numéricas destacadas:**  
  `id, year, number_of_sources, latitude, longitude, deaths_a, deaths_b, deaths_civilians, ...`
- **Variables categóricas destacadas:**  
  `conflict_name, dyad_name, side_a, side_b, source_article, source_office, source_date, region, ...`
- **Valores nulos:**  
  - Columnas con mayor porcentaje de nulos:
    - `gwnob`: 92%
    - `source_office`, `source_date`, `source_headline`: ~25%
    - `adm_2`: 18%
  - Otras columnas con pocos nulos: `where_description` 2.6%, `adm_1` 5%
- **Curiosidades:**
  - Algunas variables numéricas como `deaths_a`, `deaths_b` o `number_of_sources` presentan una cantidad significativa de outliers.

---

## Desafíos a priori para el análisis

- **Limpieza de datos:**
  - Manejo de columnas con alto porcentaje de nulos (`gwnob`, campos de fuentes como `source_office`, `source_date`).
  - Outliers en variables como `deaths_a`, `deaths_b`, `side_a_new_id` y `number_of_sources`.
  - Posibles duplicados a revisar.
- **Preparación de variables:**
  - Variables categóricas largas (`source_headline`, `source_article`) pueden requerir limpieza o normalización.
  - Posibles variables redundantes (`conflict_dset_id` vs `conflict_new_id`) a evaluar.
- **Reto adicional:**
  - Escalabilidad del análisis debido a la gran cantidad de filas (~386k).
  - Consistencia y correlaciones entre IDs numéricos y nombres categóricos (`side_a`, `side_b`, `dyad_name`).

---

💡 *Sugerencia visual para la presentación:*  
- Gráfico de barras o pastel mostrando el porcentaje de nulos por variable más relevante.  
- Boxplots de variables numéricas con outliers para ilustrar los desafíos.